src/minio/managers/policy_manager.py

```diff
@@ -198,6 +197,135 @@ async def delete_group_policy(self, group_name: str) -> None:
         """Delete a group's policy from MinIO."""
         await self.delete_policy(TargetType.GROUP, group_name)
 
+    # === USER/GROUP POLICY MANAGEMENT ===
+
+    async def create_user_policies(
+        self, username: str
+    ) -> tuple[PolicyModel, PolicyModel]:
+        """
+        Create both home and system policies for a user.
+
+        This method is idempotent - if policies already exist, it will return
+        the existing policies instead of failing.
+
+        Args:
+            username: Username to create policies for
+
+        Returns:
+            Tuple of (home_policy, system_policy)
+        """
+        async with self.operation_context("create_user_policy"):
+            # Create both policy models
+            home_policy = self._create_user_home_policy(username)
+            system_policy = self._create_user_system_policy(username)
+
+            # Check if home policy already exists
+            home_exists = await self.resource_exists(home_policy.policy_name)
+            if home_exists:
+                logger.info(
+                    f"User home policy already exists: {home_policy.policy_name}"
+                )
+                # Load existing policy to return
+                existing_home = await self._load_minio_policy(home_policy.policy_name)
+                if existing_home:
+                    home_policy = existing_home
```

Comment:
This seems odd - it checks if home exists, if it does pulls it, and then checks if the pull returned anything. Isn't it guaranteed to exist at that point?

If pulling the policy isn't way more expensive than detecting the policy, maybe just pull the policy and if you get something set home_policy

Reply:
Same for system policy below

Reply:
And group policy